massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -360,6 +361,29 @@ impl SpeculativeLedger {
         Ok(())
     }
 
+    /// Gets a copy of a datastore keys for a given address
+    ///
+    /// # Arguments
+    /// * `addr`: address to query
+    ///
+    /// # Returns
+    /// `Some(Vec<Vec<u8>>)` for found keys, `None` if the address does not exist.
+    pub fn get_keys(&self, addr: &Address) -> Option<Vec<Vec<u8>>> {
+
+        let mut added_keys = self.added_changes.get_keys(addr);
```

Comment:
Error: if a key was deleted in added_changes, that deletion won't be taken into account